src/field_theory/splitting_field.lean

```diff
@@ -739,3 +739,89 @@ end is_splitting_field
 end splitting_field
 
 end polynomial
+
+section alg_equiv_restrict
+
+open polynomial
+
+variables {F K : Type*} [field F] [field K] [algebra F K] (ϕ ψ : K →ₐ[F] K) (χ ω : K ≃ₐ[F] K)
+  (p : polynomial F) (E : Type*) [field E] [algebra F E] [algebra E K]  [is_scalar_tower F E K]
+
+lemma is_scalar_tower_to_alg_hom_range_eq_adjoin [hp : is_splitting_field F E p] :
+  (is_scalar_tower.to_alg_hom F E K).range =
+    algebra.adjoin F (↑(p.map (algebra_map F K)).roots.to_finset : set K) :=
+begin
+  rw [is_scalar_tower.algebra_map_eq F E K, ←map_map, roots_map, ←finset.image_to_finset,
+    finset.coe_image, algebra.adjoin_algebra_map, hp.adjoin_roots, algebra.map_top],
+  exact ((splits_id_iff_splits (algebra_map F E)).mpr hp.splits),
+end
+
+/-- Restrict algebra homomorphism to range -/
+def alg_hom.restrict_is_splitting_field_aux [hp : is_splitting_field F E p] :
+  (is_scalar_tower.to_alg_hom F E K).range →ₐ[F] (is_scalar_tower.to_alg_hom F E K).range :=
```

Comment:
I'm curious about the choice for the `range` of `to_alg_hom` used in this definition, since the previous lemma rewrites "away" from the range. Or is this just an auxiliary definition for `restrict_is_splitting_field` and either way it wouldn't matter much?

Reply:
It's because of `alg_equiv.of_injective` which states that an R-algebra is isomorphic to its range under an injective R-algebra homomorphism. This means that you can construct an R-algebra isomorphism of `E` by first constructing an R-algebra isomorphism of `(is_scalar_tower.to_alg_hom F E K).range`.

Reply:
(and it is just an auxiliary definition)